pkg/contexts/ocm/accessmethods/mvn/method.go

```diff
@@ -0,0 +1,408 @@
+		log = log.WithValues("file", metadata.Bin)
+		log.Debug("processing")
+		metadata.MimeType = artifact.MimeType()
+		if hash > 0 {
+			metadata.HashType = hash
+			metadata.Hash, err = getStringData(ctx, metadata.Bin+hashUrlExt(hash), fs)
+			if err != nil {
+				return nil, errors.Wrapf(err, "cannot read %s digest of: %s", hash, metadata.Bin)
+			}
+		} else {
+			log.Warn("no digest available")
+		}
+
+		// single binary dependency, this will never be a complete GAV - no maven uploader support!
+		if a.Extension != "" || singleBinary && a.Classifier != "" {
+			// in case you want to transport <packaging>pom</packaging>, then you should NOT set the extension
+			return &metadata, nil
+		}
+
+		// download the artifact into the temporary file system
+		out, err := tempFs.Create(file)
+		if err != nil {
+			return nil, err
+		}
+		defer out.Close()
+		reader, err := getReader(ctx, metadata.Bin, fs)
+		if err != nil {
+			return nil, err
+		}
+		defer reader.Close()
+
+		if hash > 0 {
+			dreader := iotools.NewDigestReaderWithHash(hash, reader)
+			_, err = io.Copy(out, dreader)
+			if err != nil {
+				return nil, err
+			}
+			sum := dreader.Digest().Encoded()
+			if metadata.Hash != sum {
+				return nil, errors.Newf("checksum mismatch for %s", metadata.Bin)
+			}
+		} else {
+			_, err = io.Copy(out, reader)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// pack all downloaded files into a tar.gz file
+	tgz, err := vfs.TempFile(fs, "", Type+"-"+artifact.FileNamePrefix()+"-*.tar.gz")
+	if err != nil {
+		return nil, err
+	}
+	defer tgz.Close()
+
+	dw := iotools.NewDigestWriterWith(digest.SHA256, tgz)
+	defer dw.Close()
+	err = tarutils.TgzFs(tempFs, dw)
+	if err != nil {
+		return nil, err
+	}
+
+	metadata.Bin = "file://" + tgz.Name()
+	metadata.MimeType = mime.MIME_TGZ
+	metadata.Hash = dw.Digest().Encoded()
+	metadata.HashType = crypto.SHA256
+	log.Debug("created", "file", metadata.Bin)
+
+	return &metadata, nil
+}
+
+func filterByClassifier(fileMap map[string]crypto.Hash, classifier string) map[string]crypto.Hash {
+	filtered := make(map[string]crypto.Hash)
+	for file, hash := range fileMap {
+		if strings.Contains(file, "-"+classifier+".") {
+			filtered[file] = hash
+		}
+	}
+	return filtered
+}
+
+func (a *AccessSpec) GavFiles(ctx accspeccpi.Context, fs ...vfs.FileSystem) (map[string]crypto.Hash, error) {
+	if strings.HasPrefix(a.Repository, "file://") && len(fs) > 0 {
+		dir := a.Repository[7:]
+		return gavFilesFromDisk(fs[0], dir)
+	}
+	return a.gavOnlineFiles(ctx)
+}
+
+func gavFilesFromDisk(fs vfs.FileSystem, dir string) (map[string]crypto.Hash, error) {
+	files, err := tarutils.ListSortedFilesInDir(fs, dir, true)
+	if err != nil {
+		return nil, err
+	}
+	return filesAndHashes(files), nil
+}
+
+// gavOnlineFiles returns the files of the Maven (mvn) artifact in the repository and their available digests.
+func (a *AccessSpec) gavOnlineFiles(ctx accspeccpi.Context) (map[string]crypto.Hash, error) {
```

Comment:
Surely there's a better way of doing this. Like instead of parsing the actual HTML node, can't we get a list of files via an API or something? There are some maven go SDKs out there.

Reply:
@Skarlso 
> There are some maven go SDKs out there.

Can you please point me to one? I haven't found a better way yet... and the documentation about maven repositories wasn't very helpful to me... Yes I would prefer a proper API, but haven't found one.

Reply:
https://github.com/raydac/mvn-golang seems okay

Reply:
> https://github.com/raydac/mvn-golang seems okay

uhm... sorry, but no

That is a maven-plugin, which allows you to build/execute go code during a maven build... but we need the opposite around, we need go executing maven commands - which I believe doesn't exist so far

Reply:
Ah sorry, I misread. You're right. That's quite the bummer.